cmd/kube-apiserver/app/server.go

```diff
@@ -133,7 +133,10 @@ func Run(s *options.APIServer) error {
 	}
 
 	// Proxying to pods and services is IP-based... don't expect to be able to verify the hostname
-	proxyTLSClientConfig := &tls.Config{InsecureSkipVerify: true}
+	proxyTLSClientConfig, err := transport.TLSConfigFor(s.KubeletConfig.TransportConfig())
```

Comment:
This is used to proxy to pods/services and nodes on ports other than the kubelet API, but not to the kubelet API. I don't think this should be using the kubelet transport config